src/rt.rs

```diff
@@ -5,8 +5,48 @@
 //! If the `runtime` feature is disabled, the types in this module can be used
 //! to plug in other runtimes.
 
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+    time::{Duration, Instant}
+};
+
+use futures_core::Future;
+
 /// An executor of futures.
 pub trait Executor<Fut> {
     /// Place the future into the executor to be run.
     fn execute(&self, fut: Fut);
 }
+
+/// A timer which provides timer-like functions, similar to tokio::time::*.
+pub trait Timer : std::fmt::Debug {
+    /// An analogue of tokio::time::sleep.
+    fn sleep(&self, duration: Duration) -> Box<dyn Sleep + Unpin>;
+
+    /// An analogue of tokio::time::sleep_until.
+    fn sleep_until(&self, deadline: Instant) -> Box<dyn Sleep + Unpin>;
+
+    /// An analogue of tokio::time::interval.
+    fn interval(&self, period: Duration) -> Box<dyn Interval>;
+}
+
+/// The generic version of tokio::time::Sleep, which itself is the output of tokio::time::sleep
+pub trait Sleep: Send + Sync + Unpin + Future<Output = ()> {
+
+    /// An analogue of tokio::time::Sleep::deadline.
+    fn deadline(&self) -> Instant;
```

Comment:
Thanks for remembering us and reaching out! (Sorry for delay!)

Fuchsia's async executor uses deadlines for its time, so that's at least good. Two points to bring up:

1. `reset` in tokio says "This function can be called both before and after the future has completed." In [fuchsia_async's Timer](https://fuchsia-docs.firebaseapp.com/rust/fuchsia_async/struct.Timer.html), this isn't true -- you can only reset a timer that's already fired. I'm also not sure what it means to reset a timer that hasn't fired because it's hard to guarantee you won't observe the signal. Does hyper reset timers that haven't fired?
2. `is_elapsed` says that it returns true when the time has elapsed, but is this the same as saying the timer has already fired?

I guess a third question: will hyper rely on these divergent behaviors? If not, perhaps we can make the documentation stricter in hyper's trait?